lib/src/configuration_interfaces/countly_config_limits.dart

```diff
@@ -0,0 +1,81 @@
+/// This class holds SDK internal limits (https://support.count.ly/hc/en-us/articles/360037753291-SDK-development-guide#01H821RTQ7AZ6J858BHP4883ZC) specific configurations to be used with CountlyConfig class and serves as an interface.
+/// You can chain multiple configurations.
+class CountlyConfigSDKInternalLimits {
+  /// private variables.
+  int _maxKeyLength = 0;
+  int _maxValueSize = 0;
+  int _maxSegmentationValues = 0;
+  int _maxBreadcrumbCount = 0;
+  int _maxStackTraceLinesPerThread = 0;
+  int _maxStackTraceLineLength = 0;
+
+  /// getters
+  int get maxKeyLength => _maxKeyLength;
+  int get maxValueSize => _maxValueSize;
+  int get maxSegmentationValues => _maxSegmentationValues;
+  int get maxBreadcrumbCount => _maxBreadcrumbCount;
+  int get maxStackTraceLinesPerThread => _maxStackTraceLinesPerThread;
+  int get maxStackTraceLineLength => _maxStackTraceLineLength;
+
+  /// setters / methods
+
+  /// Limits the maximum size of all string keys
+  ///
+  /// [keyLengthLimit] is the maximum char size of all string keys (default 128 chars)
+  CountlyConfigSDKInternalLimits setMaxKeyLength(int keyLengthLimit) {
+    if (keyLengthLimit > 0) {
```

Comment:
K said we want to avoid putting logic inside of config classes. So those lesser than checks might be moved to that part     if (config.sdkInternalLimits.maxKeyLength != 0) {
        countlyConfig['maxKeyLength'] = config.sdkInternalLimits.maxKeyLength;
      }

Reply:
fair point